File: semgrep-core/tests/patterns/js/dots_field_chaining.js
// MATCH:
foo.bar;
// MATCH:
foo.bar();
// MATCH:
foo.baz.bar;
// MATCH:
foo.baz().bar;
// MATCH:
foo.a.b.c.bar;
// MATCH:
foo.a.b().c.bar;
// MATCH:
foo.a().b().c().bar;

File: semgrep-core/tests/patterns/rust/dots_field_chaining.rs
struct A {
    x: B,
}

struct B {
    y: C,
}

struct C {
    z: (),
}

fn main() {
    let s = A { x: B { y: C { z: () } } };
    s.x;
    s.x.y;
    // ERROR: match
    s.x.y.z;
    // ERROR: match
    let () = s.x.y.z;
}
